doc/api/vm.md

```diff
@@ -1573,6 +1573,105 @@ inside a `vm.Context`, functions passed to them will be added to global queues,
 which are shared by all contexts. Therefore, callbacks passed to those functions
 are not controllable through the timeout either.
 
+## Local Worker
+
+> Stability: 1 - Experimental
+
+### Class: `LocalWorker`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Extends: {EventEmitter}
+
+A `LocalWorker` is effectively a Node.js environment that runs within the
+same thread.
+
+```mjs
+import { LocalWorker } from 'vm';
+import { fileURLToPath } from 'url';
+const w = new LocalWorker();
+const myAsyncFunction = w.createRequire(fileURLToPath(import.meta.url))('my-module');
+console.log(await myAsyncFunction());
+```
```

Comment:
I do think the docs should clarify the difference between this and a `ShadowRealm`.